R/zchunk_L131.nonco2_proc_R_S_T_Y.R

```diff
@@ -36,55 +39,168 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
+    get_data(all_data, "emissions/EDGAR_CO") %>%
+      gather(year, value, -`IPCC-Annex`, -`World Region`, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(Non.CO2 = 'CO')->
+      EDGAR_CO
+
+     ### Perform computations  ###
+
+     # First: Compute subsector share of sectoral emissions using EPA data
+
+     EPA_Ind %>%
+       left_join(EPA_tech, by = "Source_Category") %>%
+       rename(subsector = fuel) %>%
+       select(-EPA_Source_Category_Raw, -Source_Category) %>%   # Remove unnecessary columns
+       unique(.) %>%
+       gather(Non.CO2, value, -sector, -subsector) %>%  # make a new column called Non.co2
+       na.omit() %>%  # delete rows with NAs
+       group_by(sector, subsector, Non.CO2) %>% # group by sector, subsector and Non.CO2
+       summarise(x = sum(value)) -> # sum the values
+       L131.EPA_nonco2_indproc.mlt
+
+     # Second: Compute share of sectoral emissions in each subsector using EPA data
+     GCAM_sector_tech %>%
+       select(supplysector, subsector, stub.technology, EPA_agg_sector, EPA_agg_fuel_ghg, EDGAR_agg_sector) %>%
+       filter(EDGAR_agg_sector %in% c("industry_processes", "chemicals", "landfills", "wastewater","aerosols",
+                                      "metals","foams","solvents", "semiconductors")) %>%
+       repeat_add_columns(tibble::tibble(Non.CO2 = c("CH4","N2O","NMVOC","NOx","SO2","CO","VOC"))) ->
+       L131.nonco2_pct_R_prc_S_S_2005
+
+     # Third: Group by and then join by sector, subsector, and Non.CO2.  However, to join the columns they have to have
+     # the same name. Rename column to tech_emissions. Tech_emissions is the sum and tech_share is the tech_emissions/sector_emissions
+     L131.nonco2_pct_R_prc_S_S_2005 %>%
+       group_by(EPA_agg_sector, EPA_agg_fuel_ghg, Non.CO2) %>%
+       left_join(L131.EPA_nonco2_indproc.mlt, by = c(EPA_agg_sector = "sector", EPA_agg_fuel_ghg = "subsector","Non.CO2")) %>%
+       rename(tech_emissions = x) %>%
+       mutate(tech_emissions = if_else(is.na(tech_emissions),1,tech_emissions)) ->
+       L131.nonco2_pct_R_prc_S_S_2005
+
+     L131.nonco2_pct_R_prc_S_S_2005 %>%
+       group_by(EPA_agg_sector, Non.CO2) %>% # group by sector, and Non CO2
+       summarise(tech_emissions = sum(tech_emissions)) ->
+       L131.nonco2_gg_R_prc_S_2005
+
+
+     L131.nonco2_pct_R_prc_S_S_2005 %>%
+       group_by(EPA_agg_sector, Non.CO2) %>%
+       left_join(L131.nonco2_gg_R_prc_S_2005, by = c("EPA_agg_sector", "Non.CO2")) %>%
+       rename(sector_emissions = tech_emissions.y ) %>% # sector emisisons from thr join
+       rename(tech_emissions = tech_emissions.x) %>%
+       mutate(tech_share = tech_emissions/sector_emissions) ->
+       L131.nonco2_pct_R_prc_S_S_2005
+
+     # Fourth: Disaggregate EDGAR emissions to subsectors
+     # First aggregate all EDGAR data and subset for processing sectors
+
+     #filter out years 2009 and 2010 in EDGAR_VOC
+     EDGAR_VOC %>%
+       filter(year!= "2009" & year != "2010") ->
```

Comment:
`year` should be integer, not character. Also WHY are we filtering out these years?